### PR TITLE
Variables: Add the default option back for the data source variable

### LINKED ISSUE #30996: Variables: Data source variable does no longer contain a separate default option after 7.4.0 upgrade.
**What happened**:
We have got a couple of DataSources configured, one DS has been marked as `default`. This DS has been selected within a _variable_ (of the `Datasource` type) - I mean there was the something like **default** "value" among menu items and this value was selected. (The _variable_ is being referenced as a datasource for queries within a panel.)

After upgrade to `7.4.0` the `default` datasource has disappeared from the datasource _variable_, hence all data visualisations have failed. 

**What you expected to happen**:
The `defalut` "tag" on a datasource has some practical value.

**How to reproduce it (as minimally and precisely as possible)**:
1. Mark one datasource as a default one.
2. Configure a variable of the `Datasource` type within a dashboard.
3. Try to select the `default` value within this variable, there is not nothing like that.

**Anything else we need to know?**:
As a quick fix we have created a new datasource named `default`.

**Environment**: N/A
- Grafana version: 7.4.0
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by hugohaggmark**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #30996

**Special notes for your reviewer**:

### CODE REVIEW

File: public/app/features/variables/datasource/actions.ts [hunk 1]
```diff
@@ -18,10 +18,8 @@ export const updateDataSourceVariableOptions = (
   identifier: VariableIdentifier,
   dependencies: DataSourceVariableActionDependencies = { getDatasourceSrv: getDatasourceSrv }
 ): ThunkResult<void> => async (dispatch, getState) => {
-  const sources = dependencies
-    .getDatasourceSrv()
-    .getList({ metrics: true, variables: false })
-    .map(toDataSourceSelectItem);
+  const list = dependencies.getDatasourceSrv().getList({ metrics: true, variables: false });
+  const sources = list.map(toDataSourceSelectItem);
```
Comment (torkelo) [inline public/app/features/variables/datasource/actions.ts:22]: I do not think there is any value in mapping this to DataSourceSelectItem (it's sort of a deprecated type, now that DataSourcePicker handles this internally)

**Review by torkelo — COMMENTED** [review]
If possible I would skip the DataSourceSelectIem and use DataSourceInstanceSettings instead directly as a way to send data between the action and the reducer (ie the type returned by dataSourceSrv.getList  

DataSourceSelectItem should really be marked as deprecated after this PR https://github.com/grafana/grafana/pull/29542

File: public/app/features/variables/datasource/actions.ts (hunk 1 above)
Comment (hugohaggmark) [reply public/app/features/variables/datasource/actions.ts:22]: Yeah, good suggestion. I'll fix it.

Comment (hugohaggmark) [reply public/app/features/variables/datasource/actions.ts]: Fixed!

**Review by torkelo — APPROVED** [review]
Looks good

**Pull request merged**